src/main/java/mate/academy/bookstore/service/impl/OrderServiceImpl.java

```diff
@@ -0,0 +1,88 @@
+package mate.academy.bookstore.service.impl;
+
+import java.math.BigDecimal;
+import java.time.LocalDateTime;
+import java.util.Set;
+import java.util.stream.Collectors;
+import lombok.RequiredArgsConstructor;
+import mate.academy.bookstore.dto.order.CreateOrderRequestDto;
+import mate.academy.bookstore.dto.order.OrderDto;
+import mate.academy.bookstore.dto.order.UpdateOrderRequestDto;
+import mate.academy.bookstore.exception.EntityNotFoundException;
+import mate.academy.bookstore.mapper.OrderMapper;
+import mate.academy.bookstore.model.CartItem;
+import mate.academy.bookstore.model.Order;
+import mate.academy.bookstore.model.OrderItem;
+import mate.academy.bookstore.model.User;
+import mate.academy.bookstore.repository.cartitem.CartItemRepository;
+import mate.academy.bookstore.repository.order.OrderRepository;
+import mate.academy.bookstore.repository.orderitem.OrderItemRepository;
+import mate.academy.bookstore.repository.user.UserRepository;
+import mate.academy.bookstore.service.OrderService;
+import org.springframework.stereotype.Service;
+
+@Service
+@RequiredArgsConstructor
+public class OrderServiceImpl implements OrderService {
+    private final OrderRepository orderRepository;
+    private final OrderMapper orderMapper;
+    private final UserRepository userRepository;
+    private final CartItemRepository cartItemRepository;
+    private final OrderItemRepository orderItemRepository;
+
+    @Override
+    public OrderDto create(Long userId, CreateOrderRequestDto requestDto) {
+        Set<CartItem> cartItems = cartItemRepository.getCartItemsByShoppingCartId(userId);
+        if (cartItems.isEmpty()) {
+            throw new EntityNotFoundException("Can't create empty order");
+        }
+        Order order = new Order();
+        Set<OrderItem> orderItems = convertToOrderItems(order, cartItems);
+        User user = userRepository.findById(userId).orElseThrow(() ->
+                new EntityNotFoundException("Can't find user by id: " + userId));
+        order.setUser(user);
+        order.setStatus(Order.Status.PENDING);
+        order.setTotal(calculateTotalPrice(orderItems));
+        order.setOrderDate(LocalDateTime.now());
+        order.setShippingAddress(requestDto.getShippingAddress());
+        order.setOrderItems(orderItems);
+        orderRepository.save(order);
+        orderItemRepository.saveAll(orderItems);
+        return orderMapper.toDto(order);

```

Comment:
```suggestion
        Order order = orderMapper.cartToOrder(cart, orderDto.shippingAddress());
        cart.clearCart();
        return orderMapper.toOrderDto(orderRepository.save(order));
```

where in OrderMapper
```
@Mapping(target = "id", ignore = true)
@Mapping(target = "total", source = "cart.cartItems", qualifiedByName = "total")
@Mapping(target = "orderItems", source = "cart.cartItems")
Order cartToOrder(ShoppingCart cart, String shippingAddress);
```
and 
```
@AfterMapping
default void updateOrder(@MappingTarget Order order) {
    order.getOrderItems().forEach(oi -> oi.setOrder(order));
}

@Named("total")
default BigDecimal getTotal(Set<CartItem> cartItems) {
    return cartItems.stream()
                .map(i -> i.getBook().getPrice().multiply(BigDecimal.valueOf(i.getQuantity())))
                .reduce(BigDecimal.ZERO, BigDecimal::add);
}
```

And in OrderItemMapper

```
@Mapping(target = "id", ignore = true)
@Mapping(target = "price", source = "book.price")
OrderItem cartItemToOrderItem(CartItem cartItem);
```

And in OrderMapper
```
@Mapping(target = "orderDate", dateFormat = "yyyy-MM-dd HH:mm")
@Mapping(target = "userId", source = "user.id")
OrderDto toOrderDto(Order order);
```